tahoe_idp/helpers.py

```diff
@@ -60,48 +66,125 @@ def fail_if_tahoe_idp_not_enabled():
-    from Django settings.FEATURES.
-    If either value does not exist, we will raise an ImproperlyConfigured error.
+    Get API_KEY for the FusionAuth API client.
     """
     fail_if_tahoe_idp_not_enabled()
-    client_id = settings.TAHOE_IDP_CONFIGS.get("API_CLIENT_ID")
-    client_secret = settings.TAHOE_IDP_CONFIGS.get("API_CLIENT_SECRET")
+    idp_tenant_id = config_client_api.get_admin_value("IDP_TENANT_ID")
 
-    if not client_id or not client_secret:
-        raise ImproperlyConfigured(
-            "Both `API_CLIENT_ID` and `API_CLIENT_SECRET` must be present "
-            "in your `TAHOE_IDP_CONFIGS`"
-        )
+    if not idp_tenant_id:
+        raise ImproperlyConfigured("Tahoe IdP `IDP_TENANT_ID` cannot be empty in `admin` Site Configuration.")
 
-    return client_id, client_secret
+    return idp_tenant_id
 
 
-def build_auth0_query(**kwargs):
+def get_api_key():
+    """
+    Get API_KEY for the FusionAuth API client.
     """
-    Responsible for building a querystring used in Tahoe IdP GET APIs.
+    return get_required_setting("API_KEY")
 
-    This is Auth0-specific.
+
+def get_id_jwt_decode_options():
+    """
+    Get IdP jwt decode configs from Django's settings variable.
+    """
+    fail_if_tahoe_idp_not_enabled()
+    return settings.TAHOE_IDP_CONFIGS.get("JWT_OPTIONS", {})
+
+
+def get_jwt_algorithms():
+    """
+    Get
+
+    See: https://fusionauth.io/learn/expert-advice/tokens/anatomy-of-jwt
+    """
+    fail_if_tahoe_idp_not_enabled()
+    return settings.TAHOE_IDP_CONFIGS.get("JWT_ALGORITHMS", [
+        "HS256",
+        "RS256",
+        "RS384",
+        "RS512",
+        "ES256",
+        "ES384",
+        "ES512",
+    ])
+
+
+def get_api_client():
+    """
+    Get a configured Rest API client for the Identity Provider.
+    """
+    client = FusionAuthClient(
+        api_key=get_api_key(),
+        base_url=get_idp_base_url(),
+    )
+    client.set_tenant_id(get_tenant_id())
+    return client
+
+
+def get_jwt_payload(client_id, id_token):
+    """
+    Verifies and returns a JWT token payload from the response. A normal returned
+    payload looks like the following:
+        {
+            'iss': N/A,
+            'name': 'Ahmed Jazzar',
+            'email_verified': False,
+            'picture': 'https://s.gravatar.com/avatar/abcd.png',
+            'exp': 1633164125,
+            'sub': '659dc4c4-d5be-11ec-8b63-8ff1c112ea18',
+            'email': 'ahmed@appsembler.com',
+            'updated_at': '2021-10-01T22:42:54.841Z',
+            'iat': 1633128125,
+            'aud': '9TbUCXJ9F3u3Q5jyRpOWuKaybiCk3rEa'
+        }
     """
-    args = [
-        urllib.parse.quote_plus('{}:"{}"'.format(key, value))
-        for key, value in kwargs.items()
-    ]
-    return "&".join(args)
+    issuer = "{}/".format(get_idp_base_url())
+    audience = client_id
+    jwks_response = requests.get("{}/.well-known/jwks.json".format(get_idp_base_url()))
```

Comment:
timeout?

Also, do we have plans to cache this? Other OIDC implementations I've seen typically cache this for 5 minutes or so at a time since it typically won't change very often.